lib/coverband/utils/result.rb

```diff
@@ -100,6 +100,38 @@ def coverage
         Hash[keys.zip(original_result.values_at(*keys))]
       end
 
+      #
+      # Applies the configured groups to the given array of SimpleCov::SourceFile items
+      #
+      def filter_to_groups(files)
```

Comment:
I thought i had mentioned this but now i can't find my comment so going to add again. I think this method would be helped by introducing a class called FileGroups. Might be a good way to break up the logic.

Reply:
agreed as I work on tests I will break this out it is far to complex right now